### PR TITLE
Apply timeDelta and minor code improvements

### PR DESCRIPTION
**Pull request by danieldownes**
Objectives:
 - Introduce a shared deltaTime calc.
 - See the automated code-formatting in action.
 - Reduce some x/y pairs in the Camera class to vector2.
 - Slowdown the mobileAgent that follows the pathfinderAgent.
 - Remove the pathfinderAgent sprite.

I believe ensuring pathfinding for mobileAgent may be an appropriate task to discuss. I don't believe this was quite working in the PR discussed earlier. The pathfinderAgent and mobileAgent need to be synced. Currently there is no timing sync happening which causes the mobileAgent to not care about pathfinding,  and is only using pathfindingAgent as a waypoint marker.

### CODE REVIEW

File: SandSim2000/AgentManager.cpp [hunk 1]
```diff
@@ -1,16 +1,7 @@
...
-/*                        +++++++++++++ Agent Manager +++++++++++++                             */
-/* Method: UpdateImpassableTerrainNodes - Sets selected nodes as impassable terrain             */
-/* Method: PropagateWaveFrontHeuristics - Sets heuristics for nodes in a wave out from target   */
-/* Method: AStar - Performs main loop of AStar on a grid with heuristics preset from WFP        */
-/* Method: ExploreNeighbours - Neighbours of current cell passed to it                          */
-/* Method: ReconstructPath - Builds list of path nodes based on parent nodes in path            */
-/*                        +++++++++++++ Debug Methods +++++++++++++                             */
-/* Method: PrintGhostGrid - Prints the ghost grid, has commented out code to print the hScore   */
-/*             +++++++++++++     +++++++++++++        +++++++++++++                             */
-
```
Comment (danieldownes) [inline SandSim2000/AgentManager.cpp:12]: Would say this is not needed, is outdated, in the wrong place and/or is just in the way.

File: SandSim2000/Camera.cpp [hunk 2]
```diff
@@ -1,157 +1,171 @@
...
+    text1.setFont(font);
 
-    text1.setString("Mouse Grid (X: " + std::to_string(inputState.selectedCell.x) + ", Y: " + std::to_string(inputState.selectedCell.y) + ")");
+    text1.setString("Mouse Grid (X: " + std::to_string(inputState.selectedCell.x) +
+                    ", Y: " + std::to_string(inputState.selectedCell.y) + ")");
     text1.setCharacterSize(24);
-    text1.setFillColor(sf::Color::White); 
-    text1.setPosition(10, 10); 
+    text1.setFillColor(sf::Color::White);
+    text1.setPosition(10, 10);
```
Comment (danieldownes) [inline SandSim2000/Camera.cpp:168]: Our code-formating rules at play tore move white-space at the end of files. I've rarely seen this happen, so must have been some dev habbit.

File: SandSim2000/GridGenerator.cpp [hunk 3]
```diff
@@ -16,30 +16,6 @@ sf::Vector2f GridGenerator::cartesianToIsometricTransform(const sf::Vector2f& ca
...
-    const float spriteDimension = 50.0f;
-    const float invSpriteDimension = 1.0f / spriteDimension;
-
-    // Apply the inverted transformation
-    cartesianPosition.x = (isometricPosition.x / invSpriteDimension + isometricPosition.y / (2.0f * invSpriteDimension)) / 2.0f;
-    cartesianPosition.y = (isometricPosition.y / (2.0f * invSpriteDimension) - (isometricPosition.x / invSpriteDimension)) / 2.0f;
-
-    return cartesianPosition;
-}
-*/
```
Comment (danieldownes) [inline SandSim2000/GridGenerator.cpp:42]: If its important, then the dev who finds it important tag/bookmark the code in github, or save locally. We don't need useless code in the codebase.

**Review by danieldownes — COMMENTED** [review]
Added some initial comments to show my thoughts.
It may be best to see diffs on a per commit basis as there is a bit of code formatting as previously warned.

**Comment by danieldownes** [comment]
@LiamStewart2 Please can you also review, not just functionality, but the code changes if you can wade through the formatting changes..

File: SandSim2000/AgentManager.cpp (hunk 1 above)
Comment (123samueld) [reply SandSim2000/AgentManager.cpp:12]: I'm all for code which is purely self commenting, so removing is a great idea. Just I received some flakk about no "comments" so put this as a compromise. But self commenting code (clean, polite code that reads like well written pros) is superior to comments.

File: SandSim2000/BattlefieldMap.cpp [hunk 4]
```diff
@@ -12,89 +12,6 @@ void BattlefieldMap::initMap(unsigned int mapSize)
...
-    depthMap[7] = new int[size] {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
-    depthMap[8] = new int[size] {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
-    depthMap[9] = new int[size] {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
-    depthMap[10] = new int[size] {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
-    depthMap[11] = new int[size] {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
-    depthMap[12] = new int[size] {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
-    depthMap[13] = new int[size] {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
-    depthMap[14] = new int[size] {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
-    depthMap[15] = new int[size] {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
-    */
```
[the remark this answers is not in the document]
Comment (123samueld) [reply SandSim2000/BattlefieldMap.cpp:53]: Agreed. Handy to quickly switch to flat terrain but not needed most of the time.

File: SandSim2000/Camera.cpp (hunk 2 above)
Comment (123samueld) [reply SandSim2000/Camera.cpp:168]: Not entirely sure what you mean, could you elaborate a little?

File: SandSim2000/GridGenerator.cpp (hunk 3 above)
Comment (123samueld) [reply SandSim2000/GridGenerator.cpp:42]: Agreed. I wrote that and left it commented out, bad housekeeping on my part.

File: SandSim2000/Camera.cpp (hunk 2 above)
Comment (danieldownes) [reply SandSim2000/Camera.cpp:168]: Sure, you can see the end of files previously had a space at the end? Applying the code-formatting rule (VS shortcut Control-k+d chord), as defined in .clang-format file removed them. Just throught I'd point this out as I've rarely see extra spaces at the end of lines.

**Comment by danieldownes** [comment]
@123samueld This is now merged, however there is a call to console log: [INVALID SPRITE POSITION]
I don't know why, perhaps you can have a dig in?

**Comment by 123samueld** [comment]
> @123samueld I've now resolved the merge comments with main, however now there is a call to console log: [INVALID SPRITE POSITION]
> 
> I don't know why, perhaps you can have a dig in?
I'm not getting that error, can you tell me how to replicate it?

**Comment by 123samueld** [comment]
> Objectives:
> 
> * Introduce a shared deltaTime calc.
> * See the automated code-formatting in action.
> * Reduce some x/y pairs in the Camera class to vector2.
> * Slowdown the mobileAgent that follows the pathfinderAgent.
> * Remove the pathfinderAgent sprite.
> 
> I believe ensuring pathfinding for mobileAgent may be an appropriate task to discuss. I don't believe this was quite working in the PR discussed earlier. The pathfinderAgent and mobileAgent need to be synced. Currently there is no timing sync happening which causes the mobileAgent to not care about pathfinding, and is only using pathfindingAgent as a waypoint marker.

Dan, yes the mobileAgents don't directly pathfind themselves. This is intentional, because the attempt was to build an RTS about The Great War, there would be a colossal number of units and all those pathing sessions would get expensive. So instead there is a pathfinderAgent for each pathing session, whether it's just 1 unit moving or a group of 100. Then the Boids (mobileAgents) flock around the pathfinderAgent as it moves creating an organic group movement at relatively low cost. Hence why the speed of the pathfinderAgent is way too high, it will need to be better matched to the speed of it's mobileAgent, but that's for later once we have multiple mobileAgents.

**Comment by danieldownes** [comment]
> Please check my responses to 2 of your comments.
> 
> You mentioned an error "Invalid Sprite Position", I didn't get it, can you tell me how to replicate it?
> 
> Apart from that I'd approve this.

Its not an error. Its a log in the console window.

**Comment by 123samueld** [comment]
> > Please check my responses to 2 of your comments.
> > You mentioned an error "Invalid Sprite Position", I didn't get it, can you tell me how to replicate it?
> > Apart from that I'd approve this.
> 
> Its not an error. Its a log in the console window.

I'm not getting anything like that in the console, neither in MapEditor nor Skirmish modes. But there are some other comments, like "MapEditor" and what I assume is pathfinding execution time logs. Please remove all console comments before moving on.

**Review by 123samueld — CHANGES_REQUESTED** [review]
Please remove the debuggin/prototype comments ("Map Editor Selected" and numbers like 0: 0.121221 etc) then this will be ready to merge.

**Comment by danieldownes** [comment]
> Please remove the debuggin/prototype comments ("Map Editor Selected" and numbers like 0: 0.121221 etc) then this will be ready to merge.

I think that's being a bit fussy for the purpose of this PR. So I don't agree.

**Review by 123samueld — APPROVED** [review]
Thank you for applying timeDelta, that's a much better way to handle time shifts, I got complacent using SMLF's FPS limiter.

Merge when ready.

**Pull request merged**